src/logic/safe/transactions/gasNew.js

```diff
@@ -58,3 +76,75 @@ export const estimateTxGasCosts = async (
     return 10000
   }
 }
+
+export const estimateSafeTxGas = async (
+  safe: any,
+  safeAddress: string,
+  data: string,
+  to: string,
+  valueInWei: number | string,
+  operation: number,
+) => {
+  try {
+    let safeInstance = safe
+    if (!safeInstance) {
+      safeInstance = await getGnosisSafeInstanceAt(safeAddress)
+    }
+
+    const web3 = await getWeb3()
+    const estimateData = safeInstance.contract.methods.requiredTxGas(to, valueInWei, data, operation).encodeABI()
+    const estimateResponse = await web3.eth.call({
+      to: safeAddress,
+      from: safeAddress,
+      data: estimateData,
+    })
+    const txGasEstimation = new BigNumber(estimateResponse.substring(138), 16).toNumber() + 10000
+
+    // 21000 - additional gas costs (e.g. base tx costs, transfer costs)
+    const dataGasEstimation = estimateDataGasCosts(estimateData) + 21000
+
+    let additionalGasBatches = [10000, 20000, 40000, 80000, 160000, 320000, 640000, 1280000, 2560000, 5120000]
+    for (let i = 0; i < 10; i++) {
```

Comment:
Why do we need thi sfor loop still?

Reply:
So we have 10 batches? Or did I get you wrong?

Reply:
but you do `additionalGasBatches.map` which will generate the 10 batches ... this for loop only does this 10 times

Reply:
so you do `additionalGasBatches.map((gas) => gas * 2)` but that also doesn't really make sense ... since you basically just shift the elements one to the left and add the double of the last at the end. Also `5120000` is already very high. If you double that you are above the current block gas limit just with the additional gas

Reply:
So for loop is not needed, right? Sorry I misunderstood you then when we were talking in slack